common/predictive-text/worker/models/dummy-model.ts

```diff
@@ -52,11 +52,29 @@ namespace models {
       return this.configuration;
     }
 
-    predict(transform: Transform, context: Context, injectedSuggestions?: Suggestion[]): Suggestion[] {
+    predict(transform: Transform, context: Context, injectedSuggestions?: Suggestion[]): Distribution<Suggestion> {
+      let makeUniformDistribution = function(suggestions: Suggestion[]): Distribution<Suggestion> {
+        let distribution: Distribution<Suggestion> = [];
+        let n = suggestions.length;
+
+        for(let s of suggestions) {
+          distribution.push({sample: s, p: 1});  // For a dummy model, this is sufficient.  The uniformness is all that matters.
+        }
+
+        return distribution;
+      }
+
       if (injectedSuggestions) {
-        return injectedSuggestions;
+        return makeUniformDistribution(injectedSuggestions);
+      }
+
+      let currentSet = this._futureSuggestions.shift();
+      
+      if(!currentSet) {
+        return null;
```

Comment:
```suggestion
        return [];
```

Reply:
Technically, neither are perfectly accurate - the original return statement was as follows:

` return this._futureSuggestions.shift();`

When `_futureSuggestions` is an empty array (`[]`), the `.shift()` will technically return `undefined`.

Or are we saying that we want to "throw this in" as a quick additional enhancement?

Reply:
Oh, I neglected to realize the context! Since this is the _dummy_ model, I guess it doesn't super make a difference. Either throw an error (should only be thrown in tests), or an empty array — does it make sense for a model to return 0 suggestions?

Handle it by whatever means makes the most sense. Returning `null` or `undefined` is usually not useful!